apps/main/src/dex/hooks/usePoolIdByAddressOrId.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react'
+import { isAddress, isAddressEqual, type Address } from 'viem'
+import useStore from '../store/useStore'
+
+/**
+ * Tries to get a pool id from a value that is either an address or a pool id already.
+ * Might get rewritten when we refactor out the pool stores and its mappers for a tanstack query.
+ */
+export function usePoolIdByAddressOrId({ chainId, poolIdOrAddress }: { chainId: number; poolIdOrAddress: string }) {
+  const poolData = useStore((state) => state.pools.poolsMapper[chainId])
+  const poolDataCache = useStore((state) => state.storeCache.poolsMapper[chainId])
```

Comment:
This pool cache is kinda horrifying, wen tanstack refactor

Reply:
is the cache even properly used?

Reply:
Technically yeah but it's not worth the tech debt imo